images/router/haproxy/conf/haproxy-config.template

```diff
@@ -290,8 +292,8 @@ frontend fe_sni
     {{- "" }} crt-list /var/lib/haproxy/conf/cert_config.map accept-proxy
     {{- with (env "ROUTER_MUTUAL_TLS_AUTH") }}
       {{- "" }} verify {{. }}
-    {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CA") }} ca-file {{. }} {{ else }} ca-file /etc/ssl/certs/ca-bundle.trust.crt {{ end }}
-    {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CRL") }} crl-file {{. }} {{ end }}
+	{{- if $haveClientCA }} ca-file /var/lib/haproxy/mtls/latest/ca-bundle.pem {{ else }} ca-file /etc/ssl/certs/ca-bundle.trust.crt {{ end }}
+	{{- if $haveCRLs }} crl-file /var/lib/haproxy/mtls/latest/crls.pem {{ end }}
```

Comment:
You replaced spaces with tabs.  Please use spaces for consistency with the rest of the template.  

I'm wary of breaking OKD users, or people using custom router deployments (unsupported, but better not to break them if we don't need to).  If `ROUTER_MUTUAL_TLS_AUTH_CA` or `ROUTER_MUTUAL_TLS_AUTH_CRL` is set, we should still respect it.  Likewise for `ROUTER_MUTUAL_TLS_AUTH_CRL` below.  
```suggestion
    {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CA") }} ca-file {{ . }} {{- else }}{{ if $haveClientCA }} ca-file /var/lib/haproxy/mtls/latest/ca-bundle.pem {{- else }} ca-file /etc/ssl/certs/ca-bundle.trust.crt {{- end }}{{ end }}
    {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CRL") }} crl-file {{ . }} {{- else }}{{ if $haveCRLs }} crl-file /var/lib/haproxy/mtls/latest/crls.pem {{- end }}{{ end }}
```
Likewise for `ROUTER_MUTUAL_TLS_AUTH_CRL` below.

Reply:
My router changes still use `ROUTER_MUTUAL_TLS_AUTH_CA` since the operator still needs to supply the CA bundle, so using `with (env "ROUTER_MUTUAL_TRLS_AUTH_CA")` by itself to select the old behavior isn't specific enough.

What do you think of something like this?
```suggestion
    {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CRL") }} crl-file {{ . }}
        {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CA") }} ca-file {{ . }} {{- else }} ca-file /etc/ssl/certs/ca-bundle.trust.crt {{ end }}
    {{- else }}
        {{- if $haveClientCA }} ca-file /var/lib/haproxy/mtls/latest/ca-bundle.pem {{- else }} ca-file /etc/ssl/certs/ca-bundle.trust.crt {{ end }}
        {{- if $haveCRLs }} crl-file /var/lib/haproxy/mtls/latest/crls.pem {{ end }}
    {{- end}}
```

Reply:
That could work (though I think you also need to re-add `{{- with (env "ROUTER_MUTUAL_TLS_AUTH_CRL") }} crl-file {{. }} {{ end }}`).

Reply:
The fix I proposed does include `crl-file ...`, but maybe it's a bit confusing. I'll see if I can make it clearer.

Reply:
I've included a clearer version of this fix in #472.